mitmproxy/proxy/protocol/websocket.py

```diff
@@ -70,43 +71,50 @@ def _handle_data_frame(self, frame, source_conn, other_conn, is_server):
             self.flow.messages.append(websocket_message)
             self.channel.ask("websocket_message", self.flow)
 
-            def get_chunk(payload):
-                if len(payload) == length:
-                    # message has the same length, we can reuse the same sizes
-                    pos = 0
-                    for s in original_chunk_sizes:
-                        yield payload[pos:pos + s]
-                        pos += s
+            if not self.flow.stream:
+                def get_chunk(payload):
+                    if len(payload) == length:
+                        # message has the same length, we can reuse the same sizes
+                        pos = 0
+                        for s in original_chunk_sizes:
+                            yield payload[pos:pos + s]
+                            pos += s
+                    else:
+                        # just re-chunk everything into 4kB frames
+                        chunk_size = 4096
```

Comment:
Thinking about it again, we should consider the frame header as well.
So the size of each chunk should be 4096 - header.
We also have to account for the direction a frame is sent, since the masking key is only for client->server frames.
https://tools.ietf.org/html/rfc6455#section-5.2